tests/lib/rules/forbid-foreign-prop-types.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview Tests for forbid-foreign-prop-types
+ */
+'use strict';
+
+// -----------------------------------------------------------------------------
+// Requirements
+// -----------------------------------------------------------------------------
+
+var rule = require('../../../lib/rules/forbid-foreign-prop-types');
+var RuleTester = require('eslint').RuleTester;
+
+var parserOptions = {
+  ecmaVersion: 6,
+  ecmaFeatures: {
+    experimentalObjectRestSpread: true,
+    jsx: true
+  }
+};
+
+require('babel-eslint');
+
+// -----------------------------------------------------------------------------
+// Tests
+// -----------------------------------------------------------------------------
+
+var ERROR_MESSAGE = 'Using another component\'s propTypes is forbidden';
+
+var ruleTester = new RuleTester();
+ruleTester.run('forbid-foreign-prop-types', rule, {
+
+  valid: [{
+    code: 'import { propTypes } from "SomeComponent";',
+    parser: 'babel-eslint'
+  }, {
+    code: 'import { propTypes as someComponentPropTypes } from "SomeComponent";',
+    parser: 'babel-eslint'
+  }],
```

Comment:
I'm not sure why lowercase matters - I agree that cases where it's an object or array literal should be valid, but `class foo extends React.Component` should have the same restrictions (someone could always `import Foo from './foo'` and use it just fine)